### PR TITLE
Add authentication headers to enroot import command if gitlab is used as docker registry

### PR DESCRIPTION
**Pull request by staeglis**
Fix issue #9: Add authentication headers to enroot import command if gitlab is used as docker registry

### CODE REVIEW

File: prepare.sh [hunk 1]
```diff
@@ -37,7 +37,11 @@ if ! [[ $(enroot list | grep "${CONTAINER_NAME}") ]]; then
     # Import a container image from a specific location to enroot image dir
     # Scheme: docker://[USER@][REGISTRY#]IMAGE[:TAG]
     IMAGE_DIR="${ENROOT_DATA_PATH}"
-    URL="docker://${CUSTOM_ENV_CI_JOB_IMAGE}"
+    if [[ ${CUSTOM_ENV_CI_JOB_IMAGE} == "${CI_REGISTRY}"* ]]; then
```
[the remark this answers is not in the document]
Comment (staeglis) [reply prepare.sh]: Yes fixed

File: prepare.sh [hunk 2]
```diff
@@ -37,7 +37,11 @@ if ! [[ $(enroot list | grep "${CONTAINER_NAME}") ]]; then
     # Import a container image from a specific location to enroot image dir
     # Scheme: docker://[USER@][REGISTRY#]IMAGE[:TAG]
     IMAGE_DIR="${ENROOT_DATA_PATH}"
-    URL="docker://${CUSTOM_ENV_CI_JOB_IMAGE}"
+    if [[ ${CUSTOM_ENV_CI_JOB_IMAGE} == "${CUSTOM_ENV_CI_REGISTRY}"* ]]; then
+        URL="docker://${CUSTOM_ENV_CI_REGISTRY_USER}:${CUSTOM_ENV_CI_REGISTRY_PASSWORD}@${CUSTOM_ENV_CI_REGISTRY}#${CUSTOM_ENV_CI_JOB_IMAGE#*$CUSTOM_ENV_CI_REGISTRY/}"
```
Comment (tcojean) [inline prepare.sh:41]: That requires storing the registry password as a variable in either the `.gitlab-ci.yml` script or the `.gitlab-runner/config.toml` file, right? For safety, Is there a way to make this optional and rely on the `$ENROOT_CONFIG_PATH/.credentials` when the user wants to instead?

**Review by tcojean — COMMENTED** [review]
Thanks a lot for your contribution! In general, this LGTM.

Could you please add some simple documentation in the README so users know this is supported? https://github.com/ginkgo-project/gitlab-hpc-ci-cb/blob/master/README.md#global-options

File: prepare.sh (hunk 2 above)
Comment (staeglis) [reply prepare.sh:41]: No CI_REGISTRY_PASSWORD is the default token:

https://docs.gitlab.com/ee/ci/variables/

**Review by tcojean — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**